### PR TITLE
irc: /ban, /unban, /quiet, /unquiet command improvements

### LINKED ISSUE #15: Make /ban, /quiet, etc. set as many as possible at once
`RPL_ISUPPORT` (numeric `005`) can be used for this, particularly the `MODES` part. For example, `/ban a b c d e` would be split into `/ban a b c d` and `/ban e` with `MODES=4`.

### LINKED ISSUE #577: /quiet and /unquiet ignore irc.network.ban_mask_default
I noticed this while reading the logs of testing https://github.com/weechat/weechat/issues/576.

```
/quiet Euforia
2015-11-05 15:01:00+0200 -- Mode ##Mikaela [+q Euforia!*@*] by Mikaela
/unquiet Euforia
2015-11-05 15:01:04+0200 -- Mode ##Mikaela [-q Euforia!*@*] by Mikaela
```

Based on my `irc.network.ban_mask_default` `*!*@*$host` I believe that the mask for quiet should have been `*!*@*unaffiliated/mikaela/bot/dysforia` instead of `Euforia!*@*`.

Quiet is just lesser form of ban (only prevent talking and nick changes while ban also prevents joining the channel) and as it works the exact same way, I don't see why it would be immune to that option or any reason why there should be different option for it.
- Debian GNU/Linux 7.9 (wheezy)
- WeeChat 1.4-dev (git: v1.3-103-g99ebcca) Copyright (C) 2003-2015, compiled on Nov  5 2015 09:33:55

### PR DESCRIPTION
**Pull request by sim642**
This pull request provides three things:
1. /ban, /unban, /quiet, /unquiet commands will put as many masks as allowed by `MODES` into a single command, not send them one-by-one (closes #15)
2. /quiet, /unquiet follow format set by `ban_mask_default` for nicknames given as arguments (closes #577)
3. /quiet, /unquiet are only allowed on servers which support them (e.g. UnrealIRCd uses channel mode "q" for owner status, not quiet)

### CODE REVIEW

**Comment by Zarthus** [comment]
When I tested around with it everything seemed to work fine.

**Comment by jeaye** [comment]
It'd be really great to see this merged in. Any updates?

**Comment by flashcode** [comment]
I'll check and merge this pull request as soon as I'll have some time.
I know it can take lot of time to merge pull requests, but a full check with tests is not something optional for me when merging ;)
And when it is scheduled for a specific version, I'll update the milestone.

File: src/plugins/irc/irc-command.c [hunk 1]
```diff
@@ -237,6 +238,133 @@ irc_command_mode_nicks (struct t_irc_server *server,
...
+        weechat_printf (
+            NULL,
+            _("%s%s: cannot use /%s, type A channel mode \"%s\" not supported "
+              "by server"),
+            weechat_prefix ("error"), IRC_PLUGIN_NAME, command, mode);
+        return;
+    }
+
+    /* default is 4 modes max (if server did not send the info) */
+    max_modes = 4;
```
Comment (arza-zara) [inline src/plugins/irc/irc-command.c]: Maybe 3 is a better default since that's in [RFC 2812](https://tools.ietf.org/html/rfc2812#section-3.2.3).

Comment (Zarthus) [inline src/plugins/irc/irc-command.c]: Not many IRCds (basically none) follow RFC2812 (at least strictly to the letter), and the server is very likely to send this information. I think 4 is a good default, personally.

Comment (sim642) [inline src/plugins/irc/irc-command.c]: The general code for this function has been taken from another already existing function where it is as such: https://github.com/weechat/weechat/blob/34ccf3fd81c6c48cbfa431d218e577c6e4b39d7c/src/plugins/irc/irc-command.c#L105-L106.

**Pull request merged**